content/en/blog/2022/k8sattributes.md

```diff
@@ -0,0 +1,261 @@
+  name: sidecar-for-my-app
+spec:
+  mode: sidecar
+  image: ghcr.io/open-telemetry/opentelemetry-collector-releases/opentelemetry-collector-contrib:0.47.0
+  config: |
+    receivers:
+      jaeger:
+        protocols:
+          grpc:
+          thrift_binary:
+          thrift_compact:
+          thrift_http:
+      otlp:
+        protocols:
+          grpc:
+          http:
+
+    processors:
+      resourcedetection:
+        detectors: [env]
+        timeout: 2s
+        override: false
+
+    exporters:
+      jaeger:
+        endpoint: jaeger-all-in-one-collector:14250
+        tls:
+          insecure: true
+
+    service:
+      pipelines:
+        traces:
+          receivers: [otlp, jaeger]
+          processors: [resourcedetection]
+          exporters: [jaeger]
+```
+
+And then deploy the vertex app example, you can see the
+`OTEL_RESOURCE_ATTRIBUTES` environment variable gets injected with some values
+in the sidecar container. Some of them use the Kubernetes downward API to get
+the attribute values.
+
+Here's an example of the value of the environment variable:
+
+```yaml
+- name: OTEL_RESOURCE_ATTRIBUTES
+  value: k8s.deployment.name=dep-vertex,k8s.deployment.uid=ef3fe26b-a690-4746-9119-d2dbd94b469f,
+  k8s.namespace.name=default,k8s.node.name=$(OTEL_RESOURCE_ATTRIBUTES_NODE_NAME),k8s.pod.name=
+  (OTEL_RESOURCE_ATTRIBUTES_POD_NAME),k8s.pod.uid=$(OTEL_RESOURCE_ATTRIBUTES_POD_UID),k8s.replicaset
+  name=dep-vertex-59b6f76585,k8s.replicaset.uid=5127bc38-e298-40e1-95df-f4a777e3176c
+```
+
+## Conclusion
+
+This post covers how to configure the OpenTelemetry collector to attach
+Kubernetes resource metadata as resource attributes to OpenTelemetry traces. The
+scenarios covered, although basic, illustrate how to add this kind of metadata
+to traces so that you can incorporate the technique into other more
+sophisticated scenarios. If you want to learn more about different scenarios or
+options for configure the processors you can see the K8sattributes processor
+documentation where you can find more scenarios like sidecar, or when one
+collector as an agent report to another collector.
+
+## References
+
+- [K8sattributes processor](https://pkg.go.dev/github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sattributesprocessor)
+- [K8sattributes processor RBAC](https://pkg.go.dev/github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sattributesprocessor#hdr-RBAC)
+- [OpenTelemetry Kubernetes attributes](/docs/reference/specification/resource/semantic_conventions/k8s)
+- [resource detector processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/resourcedetectionprocessor/README.md)
+
+[pr#832]: https://github.com/open-telemetry/opentelemetry-operator/pull/832
+[opentelemetry operator]:
+  https://github.com/open-telemetry/opentelemetry-operator
+[kubernetes resource semantic conventions]:
+  /docs/reference/specification/resource/semantic_conventions/k8s
```

Comment:
@rubenvp8510 and @cartermp : what do you think of the suggestion below as an alternative to a "conclusion"

```suggestion
## Learn more

This post covers how to configure the OpenTelemetry collector to attach
Kubernetes resource metadata as resource attributes to OpenTelemetry traces. The
scenarios covered, although basic, illustrate how to add this kind of metadata
to traces so that you can incorporate the technique into other more
sophisticated scenarios. If you want to learn more about different scenarios or
options for configure the processors you can see the [K8sattributes processor
documentation][] where you can find more scenarios like sidecar, or when one
collector as an agent report to another collector.

## References

- [K8sattributes processor](https://pkg.go.dev/github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sattributesprocessor)
- [K8sattributes processor RBAC](https://pkg.go.dev/github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sattributesprocessor#hdr-RBAC)
- [OpenTelemetry Kubernetes attributes](/docs/reference/specification/resource/semantic_conventions/k8s)
- [Resource detector processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/resourcedetectionprocessor/README.md)

[pr#832]: https://github.com/open-telemetry/opentelemetry-operator/pull/832
[opentelemetry operator]:
  https://github.com/open-telemetry/opentelemetry-operator
[k8sattributes processor documentation]:
  https://pkg.go.dev/github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sattributesprocessor
[kubernetes resource semantic conventions]:
  /docs/reference/specification/resource/semantic_conventions/k8s
```

Reply:
I like it.